src/dv360.ts

```diff
@@ -190,4 +191,84 @@ export class DV360 extends ApiClient {
     }
     return campaigns;
   }
+
+  protected createSdfDownloadOperation(advertiserId: string) {
+    const downloadOperation = this.fetchEntity(
+      this.getUrl('sdfdownloadtasks'),
+      {
+        method: 'post'
+      },
+      {
+        version: 'SDF_VERSION_5_5',
+        advertiserId: advertiserId,
+        parentEntityFilter: {
+          fileType: [
+            'FILE_TYPE_CAMPAIGN',
+            'FILE_TYPE_INSERTION_ORDER',
+            'FILE_TYPE_LINE_ITEM',
+            'FILE_TYPE_AD_GROUP',
+            'FILE_TYPE_AD'
+          ],
+          filterType: 'FILTER_TYPE_NONE',
+          filterIds: []
+        }
+      }
+    );
+    return downloadOperation;
+  }
+
+  protected waitForDownloadOperationResource(downloadOperationName: string) {
+    const operationResourceUrl = this.getUrl(downloadOperationName);
+    const initialDelay = 2000;
+    const maxRetries = 10;
+    const delayMultiplier = 2;
+
+    let downloadOperation;
+    let delay = initialDelay;
+    let tryCount = 0;
+    while (tryCount <= maxRetries) {
+      downloadOperation = this.fetchEntity(operationResourceUrl);
+      Logger.log(downloadOperation);
+      if (downloadOperation.done) {
+        return downloadOperation.response.resourceName;
+      }
+      Logger.log(`Backing off for ${delay}ms`);
+      Utilities.sleep(delay);
+      tryCount++;
+      delay *= delayMultiplier;
+    }
+    Logger.log(`Try limit exceeded after ${tryCount} tries`);
+    return undefined;
+  }
+
+  downloadMedia(resourceName: string): Object {
+    //url doesn't contain version, just /media/
+    const downloadMediaUrl = this.getUrl(
+      `download/${resourceName}?alt=media`
+    ).replace('/v2', '');
+    const downloadMedia = this.fetchBlob(downloadMediaUrl);
+    //TODO move somewhere else afterwards
+    downloadMedia.setContentType('application/zip');
+    var unZippedfiles = Utilities.unzip(downloadMedia);
+    unZippedfiles.forEach((blob) => {
+      const fileName = blob.getName();
+      const values = Utilities.parseCsv(blob.getDataAsString());
+      Logger.log(`Putting ${values.length} rows into sheet ${fileName}`);
+      const targetSheet = SheetUtils.getOrCreateSheet(fileName);
```

Comment:
I would prefer not writing to the sheet but returning those values instead (just to separate the logic: dv360 class should not know anything about sheets or processing the SDF values).
E.g. ` downloadMedia(resourceName: string): Array<Array<string>> `